src/sles.py

```diff
@@ -0,0 +1,70 @@
+import json
+import urllib.request
+
+from bs4 import BeautifulSoup
+from datetime import datetime
+
+
+PRODUCT = "sles"
+URL = "https://www.suse.com/lifecycle"
+
+
+def fetch_releases(url):
+    headers = {"user-agent": "mozilla"}
+    req = urllib.request.Request(url, headers=headers)
+    with urllib.request.urlopen(req, timeout=5) as response:
+        return BeautifulSoup(response, features="html5lib")
+
+
+def convert_date(date_str):
+    return datetime.strptime(date_str, "%d %b %Y").strftime("%Y-%m-%d")
+
+
+def strip_version(version_str):
+    return version_str.strip("SUSE Linux Enterprise Server ")
```

Comment:
Having versions without spaces is better:

```suggestion
    return version_str.strip("SUSE Linux Enterprise Server ").replace(' ', '-')
```